src/lib/classes/lsp3-universal-profile.spec.ts

```diff
@@ -181,7 +152,83 @@ describe('LSP3UniversalProfile', () => {
 
       const [result] = await universalProfile.connect(signers[0]).callStatic.getData([key]);
       const checkedsumResult = ethers.utils.getAddress(result);
-      expect(checkedsumResult).toEqual(secondController);
+      expect(checkedsumResult).toEqual(secondControllerAddress);
+    });
+
+    it.only('first account should be able to setData', async () => {
+      const abi = await universalProfile.populateTransaction.setData(
+        ['0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5'],
+        ['0xbeefbeef']
+      );
+
+      const result = await keyManager.connect(signers[0]).execute(abi.data);
+
+      expect(result).toBeTruthy();
+
+      const data = await universalProfile.getData([
+        '0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5',
+      ]);
+
+      expect(data).toEqual(['0xbeefbeef']);
+    });
+    it.only('second account should be able to setData', async () => {
+      const abi = await universalProfile.populateTransaction.setData(
+        ['0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5'],
+        ['0xcafecafe']
+      );
+
+      const result = await keyManager.connect(signers[1]).execute(abi.data);
+
+      expect(result).toBeTruthy();
+
+      const data = await universalProfile.getData([
+        '0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5',
+      ]);
+
+      expect(data).toEqual(['0xcafecafe']);
+    });
+  });
+  it('should deploy reactive', (done) => {
+    const deployments$ = lspFactory.LSP3UniversalProfile.deploy(
+      {
+        controllerAddresses: [signers[0].address],
+        lsp3Profile: lsp3ProfileJson,
+      },
+      {
+        deployReactive: true,
+      }
+    ) as Observable<DeploymentEvent>;
+
+    let erc725Address: string;
+    let keyManagerAddress: string;
+
+    deployments$.subscribe({
+      next: (deploymentEvent: DeploymentEvent) => {
+        if (
+          deploymentEvent.receipt?.contractAddress &&
+          deploymentEvent.contractName === 'ERC725Account'
+        ) {
+          erc725Address = deploymentEvent.receipt.contractAddress;
+        }
+
+        if (
+          deploymentEvent.receipt?.contractAddress &&
+          deploymentEvent.contractName === 'KeyManager'
+        ) {
+          keyManagerAddress = deploymentEvent.receipt.contractAddress;
+        }
+      },
+      error: () => {
+        done();
+      },
```

Comment:
do you really want to call `done()` on error?

If the deployment fails, will it go here, and call done, and not test anything?